query/src/main/java/org/infinispan/query/impl/EntityLoader.java

```diff
@@ -23,20 +25,29 @@ public EntityLoader(AdvancedCache<?, ?> cache, KeyTransformationHandler keyTrans
       this.cache = cache;
    }
 
+   private Object decodeKey(EntityInfo entityInfo) {
+      return keyTransformationHandler.stringToKey(entityInfo.getId().toString(), cache.getClassLoader());
+   }
+
    public Object load(EntityInfo entityInfo) {
-      Object cacheKey = keyTransformationHandler.stringToKey(entityInfo.getId().toString(), cache.getClassLoader());
-      return cache.get(cacheKey);
+      return cache.get(decodeKey(entityInfo));
    }
 
-   public List<Object> load(Collection<EntityInfo> entityInfos) {
-      ArrayList<Object> list = new ArrayList<Object>(entityInfos.size());
+   public List<Object> load(List<EntityInfo> entityInfos) {
+      int entitiesSize = entityInfos.size();
+      Set<Object> keys = new LinkedHashSet<>(entitiesSize);
       for (EntityInfo e : entityInfos) {
-         Object entity = load(e);
-         if (entity != null) {
-            list.add(entity);
+         keys.add(decodeKey(e));
+      }
+      Map<?, ?> entries = cache.getAll(keys);
+      List<Object> results = new LinkedList<>();
+      for (Object key : keys) {
```

Comment:
@wburns , sorry not following.
Above you mentioned that you _could_ return the an ordered map but you'd get in trouble with the null elements. We don't need those null elements, so that would be ok?

No need to rush such a thing though, just trying to understand.